baseapp/baseapp_payments/utils.py

```diff
@@ -190,3 +208,42 @@ def list_products(self, **kwargs):
         except Exception as e:
             logger.exception(e)
             raise Exception("Error retrieving products in Stripe")
+
+    def retrieve_product(self, product_id):
+        try:
+            product = stripe.Product.retrieve(product_id, expand=["default_price"])
+            return product
+        except stripe.error.InvalidRequestError as e:
+            logger.error(f"Failed to retrieve product: {str(e)}")
+            return None
+        except stripe.error.StripeError as e:
+            logger.error(f"Stripe API error: {str(e)}")
+            return None
+
+    def list_payment_methods(self, customer_id, type="card"):
+        try:
+            payment_methods = stripe.PaymentMethod.list(
+                customer=customer_id,
+                type=type,
+            )
+            return payment_methods.data
```

Comment:
We should probably be piping the data through our own serializer, returning exactly what we get back from stripe gives us a lot less control of our endpoint response.